### ISSUE TITLE
Suggested Feature:  In app info, say if it's a xapk or a regular apk

### ISSUE THREAD

**Issue by JordanFriendshuh** [facundoolano/google-play-scraper#405]
I would really appreciate a feature in the app details that says if the apk found on gplay is a regular apk or if it's an xapk.  The hardware I am working with doesn't work wtih xapks.

Hopefully this isn't a terribly hard feature.

**Comment by icarcal** [comment]
@JordanFriendshuh Is this information available at the gplay store app page?

**Comment by gamebenchjake** [comment]
I don't believe Google Play offers XAPK downloads, as all it is is a compressed directory with the APK and an OBB file. The XAPK was made to allow clean installs of apps downloaded off of sites OUTSIDE of Google Play.


@icarcal After looking through the raw API responses from the batchExecute endpoint, doesn't look like there's any reference to whether it's an xapk or not which is in line with my above statement.

@JordanFriendshuh here's some Typescript code with the logic to check if the file is a regulaar apk or xapk:

```
export function getApkType (apkPath: string): ApkType {
    const zip = new Zip(apkPath);
    const entries = zip.getEntries();
    const entryNames = entries.map(entry => entry.entryName);
    if (_.includes(entryNames, 'manifest.json')) {
        return 'xapk';
    } else if (_.includes(entryNames, 'AndroidManifest.xml')) {
        return 'apk';
    } else {
        throw new Error(`${apkPath} is not a valid apk`);
    }
}
```

There shouldn't be a hardware limitation on the use of XAPKs... You can look at https://github.com/cubehouse/xapkinstall for some inspiration

**Comment by facundoolano** [comment]
Issue automatically closed. Please create a new one if the issue persists
